idl/chimera.pro

```diff
@@ -0,0 +1,535 @@
+;+
+; Project     : Coronal Hole Identification via Multi-thermal Emission Reconition Algorithm (CHIMERA)
+;
+; Name        : chimera
+;
+; Purpose     : Generate a coronal hole segmented tricolour image and corresponding property .txt file
+;
+; Syntax      : chimera
+;
+; Inputs      : 171A .fits file
+;		193A .fits file
+;		211A .fits file
+;		hmi .fits file
+;
+; Outputs     : chimera.png
+;	      : chimera.txt
+;
+; Keywords    : temp= input directory
+;		outpath= output directory
+;		track= directory containing tracking information
+;
+; Example    : IDL> chimera,temp='location/string/', outpath='location/string/', track='location/string/'
+;
+;
+; History     : Written 01-jun-2016, Tadhg Garton, TCD
+;
+; Contact     : gartont@tcd.ie
+;		info@solarmonitor.org
+;
+;-
+
+pro chimera,TEMP=temp,OUTPATH=outpath,TRACK=track
+
+;=============defines location of .fits files=======================
+if keyword_set(outpath) then outpath=outpath else cd, current=outpath
+if keyword_set(temp) then temp=temp else cd, current=temp
+loadct,39,/silent
+
+;==============Finds all fits files==============
+f171=findfile(temp+'/AIAsynoptic0171.f*')
+f193=findfile(temp+'/AIAsynoptic0193.f*')
+f211=findfile(temp+'/AIAsynoptic0211.f*')
+fhmi=findfile(temp+'/HMI*mag.f*')
+
+if f171 eq '' or f193 eq '' or f211 eq '' or fhmi eq '' then goto, jump1
+fil=strarr(3)
+fil[0]=f171
+fil[1]=f193
+fil[2]=f211
+
+;============set plots for z buffer=======================
+set_plot,'z'
+!p.color=0
+!p.background='FFFFFF'xL
+Device, Set_Resolution=[1588,1588], Decomposed=1, Set_Pixel_Depth=24, set_font='helvetica'
+
+;=====Reads in data=====
+read_sdo,fhmi,hin,hd
+read_sdo,fil,ind,data
+
+;=====Rotates magnetogrames if necessary======
+if hin.crota2 gt 90 then hd=rotate(temporary(hd),2)
+
+;=====Attempts to verify data is level 1.5=====
+;if ind[1].lvl_num ne 1.5 then begin
+;	aia_prep,fil,-1,ind,data
+;endif
+
+;=====Resize and smooth image=====
+data=float(data)
+data=rebin(data,1024,1024,3)
+
+;=====Alternative coordinate systems=====
+wcs=fitshead2wcs(ind[1])
+
+ind[*].naxis1=4096  
+ind[*].naxis2=4096  
+
+data=rebin(data,4096,4096,3)
+s = size(data)
+
+coord=wcs_get_coord(wcs)
+coord=rebin(coord,2,s[1],s[2])
+wcs_convert_from_coord,wcs,coord,'hg',lon,lat
+xco=coord[0,*,*]
+yco=coord[1,*,*]
+
+;=======setting up arrays to be used============
+ident=1
+iarr=bytarr(s[1],s[2])
+offarr=iarr
+mas=fltarr(s[1],s[2])
+mak=fltarr(s[1],s[2])
+msk=fltarr(s[1],s[2])
```

Comment:
That's correct,  `boolarr` would only work on 8 and above. I suppose you won't get much performance difference with other stuff.